r/R/dataset-format.R

```diff
@@ -113,6 +115,45 @@ ParquetFileFormat$create <- function(...,
 #' @export
 IpcFileFormat <- R6Class("IpcFileFormat", inherit = FileFormat)
 
+#' JSON dataset file format
+#'
+#' @description
+#' A `JsonFileFormat` is a [FileFormat] subclass which holds information about how to
+#' read and parse the files included in a JSON `Dataset`.
+#'
+#' @section Factory:
+#' `JsonFileFormat$create()` can take options in the form of lists passed through as `parse_options`,
+#'  or `read_options` parameters.
+#'
+#'  Available `read_options` parameters:
+#'  * `use_threads`: Whether to use the global CPU thread pool. Default `TRUE`. If `FALSE`, JSON input must end with an
+#'  empty line.
+#'  * `block_size`: Block size we request from the IO layer; also determines size of chunks when `use_threads`
+#'   is `TRUE`.
```

Comment:
This leads me to wonder what determines the size of chunks when `use_threads` is false?  Although maybe there just aren't chunks?

Reply:
I have no idea; I copied from https://github.com/apache/arrow/blob/95972cca1f65179b63a9fe5262d1092547e3d137/r/R/csv.R#L378-L384

Reply:
Good enough for me :)